keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/DifferentiatorTest.java

```diff
@@ -42,10 +40,10 @@ public void canReverseAutoDiffOfMultiplicationWithSingleOutputWithRespectToMany(
         DoubleVertex B = new GaussianVertex(0, 1);
         DoubleVertex C = A.times(B);
 
-        PartialDerivatives dC = Differentiator.reverseModeAutoDiff(C, ImmutableSet.of(A, B));
+        PartialsOf dC = Differentiator.reverseModeAutoDiff(C, ImmutableSet.of(A, B));
 
-        DoubleTensor dCdA = dC.withRespectTo(A);
-        DoubleTensor dCdB = dC.withRespectTo(B);
+        DoubleTensor dCdA = dC.withRespectTo(A).get();
+        DoubleTensor dCdB = dC.withRespectTo(B).get();
```

Comment:
Bit ugly we have to call this get() method all over the place?

Reply:
If you want it as a DoubleTensor then yes. We could call it `toDoubleTensor()` or we could just return the DoubleTensor from the `withRespectTo(...)` method. It doesn't look like the `withRespectTo(...)` method is ever used to get the PartialDerivative.

Reply:
Having `withRespectTo(...)` return a DoubleTensor removes all of these `get()` calls. Nice clean up.